sidecar/go.mod

```diff
@@ -0,0 +1,174 @@
+module github.com/minio/operator/sidecar
+
+go 1.22.2
```

Comment:
In `.github/workflows/kubernetes-tests.yml` it uses Go 1.21.

Reply:
Let's keep this `go 1.21`